### PR TITLE
Reassign ACOs automatically from 271s

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/controllers/warehouse_reports/health/eligibility_results_controller.rb [hunk 1]
```diff
@@ -14,13 +14,25 @@ def show
...
       @inquiry.managed_care_ids.include? patient.medicaid_id
     end
     helper_method :has_managed_care
 
+    def aco_changes
+      ids = []
+      referral_scope.where(medicaid_id: @inquiry.eligible_ids).each do |referral|
+        medicaid_id = referral.medicaid_id
+        aco_id = Health::AccountableCareOrganization.active.find_by(edi_name: @inquiry.aco_names[medicaid_id])&.id
+        ids << medicaid_id unless aco_id == referral.accountable_care_organization_id
```
Comment (eanders) [inline app/controllers/warehouse_reports/health/eligibility_results_controller.rb:31]: Do we need to catch blanks?

File: app/controllers/warehouse_reports/health/eligibility_results_controller.rb [hunk 2]
```diff
@@ -14,13 +14,28 @@ def show
...
       @inquiry.managed_care_ids.include? patient.medicaid_id
     end
     helper_method :has_managed_care
 
+    def aco_changes
+      ids = []
+      referral_scope.where(medicaid_id: @inquiry.eligible_ids).each do |referral|
+        medicaid_id = referral.medicaid_id
+        edi_name = @inquiry.aco_names[medicaid_id]
+        if edi_name.present?
```
Comment (eanders) [inline app/controllers/warehouse_reports/health/eligibility_results_controller.rb:31]: don't you mean `next unless edi_name.present?`  Rubocop isn't going to like this.

File: app/jobs/health/flag_ineligible_patients_job.rb [hunk 3]
```diff
@@ -38,8 +38,10 @@ def perform(inquiry_id)
           patient.update!(coverage_level: Health::Patient.coverage_level_managed_value,
                           previous_aco_name: previous_aco,
                           aco_name: aco)
-          aco_id = Health::AccountableCareOrganization.active.find_by(edi_name: aco)&.id
-          patient.patient_referral.update!(accountable_care_organization_id: aco_id) if aco_id.present?
+          if aco.present?
+            aco_id = Health::AccountableCareOrganization.active.find_by(edi_name: aco)&.id
+            patient.patient_referral.update!(accountable_care_organization_id: aco_id) if aco_id.present?
```
Comment (eanders) [inline app/jobs/health/flag_ineligible_patients_job.rb:43]: This is already present on line 43 (trailing if)

Comment (awisema) [reply app/jobs/health/flag_ineligible_patients_job.rb:43]: They are slightly different, line 41 is intended to capture the case where the ACO is blank in the 271, and the trailing if is for the case where the ACO appears in the 271, but we have no mapping...

I was worrying about the case where there is no ACO in the 271, but there is blank EDI name in the table...

Comment (eanders) [reply app/jobs/health/flag_ineligible_patients_job.rb:43]: Ah!

**Pull request merged**